pandas/core/indexes/base.py

```diff
@@ -5557,6 +5557,30 @@ def _should_fallback_to_positional(self) -> bool:
         missing : np.ndarray[np.intp]
             An indexer into the target of the values not found.
             These correspond to the -1 in the indexer array.
+
+        Examples
+        --------
+        >>> index = pd.Index(['c', 'b', 'a', 'b', 'b'])
+        >>> index.get_indexer_non_unique(['b', 'b'])
+        (array([1, 3, 4, 1, 3, 4]), array([], dtype=int64))
+
+        In the example below there are no matched values. For this reason,
+        the returned ``indexer`` contains only integers equal to -1.
+        It demonstrates no index at these positions that match the corresponding
+        ``target`` values. The mask [0, 1, 2] in the return value shows that the first,
+        second, and third elements are missing.
+
+        >>> index = pd.Index(['c', 'b', 'a', 'b', 'b'])
+        >>> index.get_indexer_non_unique(['q', 'r', 't'])
+        (array([-1, -1, -1]), array([0, 1, 2]))
+
+        >>> index = pd.Index(['c', 'b', 'a', 'b', 'b'])
+        >>> index.get_indexer_non_unique(['f', 'b', 's'])
+        (array([-1,  1,  3,  4, -1]), array([0, 2]))
+
+        Notice that the return value is a tuple contains two items. In the example
+        above the first item is an array of locations in ``index``. The second
+        item is a mask shows that the first and third elements are missing.
```

Comment:
shouldn't this come first?

So, overall, I'd suggest:

1. the example with `>>> index.get_indexer_non_unique(['f', 'b', 's'])`...
2. this text "Notice that the return value is a tuple", then the example with `>>> index.get_indexer_non_unique(['f', 'b', 's'])`
3. The text "In the example below there are no matched values."and then the example with `['q', 'r', 't']`

Reply:
Thank you, MarcoGorelli, for your suggestions. I updated my PR accordingly.